Mathlib/Data/SetLike/Basic.lean

```diff
@@ -6,6 +6,7 @@ Authors: Eric Wieser
 import Mathlib.Tactic.Monotonicity.Attr
 import Mathlib.Tactic.SetLike
 import Mathlib.Data.Set.Basic
+import Mathlib.Order.Atoms
```

Comment:
We should import SetLike.Basic into Order.Atoms instead.
Also we would need OrderedSetLike if #20638 gets merged.
Can you golf `Ideal.isMaximal_iff` too?

Reply:
I am trying to add `SetLike.covBy_iff` and `SetLike.isAtom_iff` by symmetry.

Reply:
(for isAtom, the same proof works, I'm struggling a little bit with covBy)

Reply:
I added 2 similar `SetLike.covBy_iff` lemmas, but their proofs are crying for some golfing, and why not deleting them.